### PR TITLE
Removed beauty at the request of Rohesie

### PR DESCRIPTION
**Pull request by Watermelon914**
## About The Pull Request

Removes beauty element and anything associated with it at the request of Rohesie.
image

## Why It's Good For The Game

Beauty code is pretty terrible, it's an element that tries to be a component by storing states. Causes a lot of runtimes, takes up a lot of memory and is generally unperformant.

## Changelog
:cl:
del: Removed beauty from the game.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
The poisoned apple has been planted.

**Comment by Mickyan** [comment]
You should remove the snob quirk also

**Comment by LemonInTheDark** [comment]
I don't think it runtimes all that much, but I'm not really sure it's worth keeping, since we didn't really go as hard on mood as we maybe should have

**Comment by RaveRadbury** [comment]
> You should remove the snob quirk also as it doesn't do anything without beauty

I'm sad to see it go. Hopefully it can come back some day.

**Comment by Mothblocks** [comment]
Can you expand on beauty code being bad? I briefly saw the discussion about how it doesn't properly handle references, but there are plenty of ways to get around that--we have weakrefs and del signals already.

> it's an element that tries to be a component by storing states

This is like, half our elements, it's a system to save lots of memory. 

I think the beauty system is nice, and I think it's a good example of what mood can be with design that isn't "I read a book".

File: code/game/objects/structures/statues.dm [hunk 1]
```diff
@@ -9,7 +9,6 @@
 	CanAtmosPass = ATMOS_PASS_DENSITY
 	material_modifier = 0.5
 	material_flags = MATERIAL_AFFECT_STATISTICS
-	/// Beauty component mood modifier
 	var/impressiveness = 15
```
Comment (Mothblocks) [inline code/game/objects/structures/statues.dm:12]: You're only removing the doc comment, not the variable.

**Comment by Rohesie** [comment]
> 
> 
> Can you expand on beauty code being bad? I briefly saw the discussion about how it doesn't properly handle references, but there are plenty of ways to get around that--we have weakrefs and del signals already.
> 
> > it's an element that tries to be a component by storing states
> 
> This is like, half our elements, it's a system to save lots of memory.
> 
> I think the beauty system is nice, and I think it's a good example of what mood can be with design that isn't "I read a book".

The problem is in `area_sensitive_contents` and most of what it touches:

```dm
/area/Entered(atom/movable/M)
	set waitfor = FALSE
	SEND_SIGNAL(src, COMSIG_AREA_ENTERED, M)
	for(var/atom/movable/recipient as anything in M.area_sensitive_contents)
		SEND_SIGNAL(recipient, COMSIG_ENTER_AREA, src)
	(...)
```

```dm
/area/Exited(atom/movable/M)
	SEND_SIGNAL(src, COMSIG_AREA_EXITED, M)
	for(var/atom/movable/recipient as anything in M.area_sensitive_contents)
		SEND_SIGNAL(recipient, COMSIG_EXIT_AREA, src)
```

```dm
/atom/movable/Exited(atom/movable/AM, atom/newLoc)
	. = ..()
	if(AM.area_sensitive_contents)
		for(var/atom/movable/location as anything in get_nested_locs(src) + src)
			LAZYREMOVE(location.area_sensitive_contents, AM.area_sensitive_contents)

/atom/movable/Entered(atom/movable/AM, atom/oldLoc)
	. = ..()
	if(AM.area_sensitive_contents)
		for(var/atom/movable/location as anything in get_nested_locs(src) + src)
			LAZYADD(location.area_sensitive_contents, AM.area_sensitive_contents)

/// See traits.dm. Use this in place of ADD_TRAIT.
/atom/movable/proc/become_area_sensitive(trait_source = TRAIT_GENERIC)
	if(!HAS_TRAIT(src, TRAIT_AREA_SENSITIVE))
		RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_AREA_SENSITIVE), .proc/on_area_sensitive_trait_loss)
		for(var/atom/movable/location as anything in get_nested_locs(src) + src)
			LAZYADD(location.area_sensitive_contents, src)
	ADD_TRAIT(src, TRAIT_AREA_SENSITIVE, trait_source)
```

That's not actually a lazy list, by the byes.
And this is necessary for the area beauty element/component to work.
The whole feature needs to be re-written better, really.

**Comment by LemonInTheDark** [comment]
It works off move signals, so it works fine with harddels most of the time. There were some cases of like, stacking, those are solved by #59371, there were also some that were caused by the sensitive object holding itself in its own list, but I solved those a while back. I don't think there's any other harddels left with it or the beauty component #57736

What runtimes are you talking about exactly?

**Comment by Rohesie** [comment]
@AnturK was quoting runtimes from iterating `area_sensitive_contents`, which contained nulls.
Since it doesn't listen to the destruction of the things it contains, it's natural to accumulate hard deletions.
And that's just one of the issues.

**Comment by LemonInTheDark** [comment]
I'm not aware of any harddels linked to the area_sensitive contents list, at least post that big fix pr I made. It doesn't need to listen for deletion since deleted objects will move to nullspace, which will remove them from the area_sensitive_contents list. There was an error linked to stuff becoming area sensitive twice, which would cause harddels, but that's fixed now

**Comment by Mothblocks** [comment]
I like beauty as a feature, so I want to wait a bit to see if the runtimes come back after Lemon's PR.

**Comment by LemonInTheDark** [comment]
For the record I'm more then willing to just keep fixing harddels with them, might catch some other ones along the way

**Comment by Mothblocks** [comment]
That's what I'd rather do, yes. 😄

**Comment by Rohesie** [comment]
Looks like the issues I was seeing were a bit exaggerated.
I still don't like the loops to send signals nor the wrongly-used lazy list macro usage, but it doesn't warrant a full removal.

**Pull request closed without merging**